### PR TITLE
Animation drop down sections

### PR DESCRIPTION
**Pull request by dkmityuk**

### CODE REVIEW

File: Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import UIKit
+
+class TaskDateHeader: UITableViewHeaderFooterView {
```
Comment (Lomovsky-Alex) [inline Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift]: `final`
Please, mark all classes as final

File: Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            }
+        }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+
+    func setUp(object: SectionModel) {
+        dateLabel.text = object.title
+        showTasksButton.setImage(UIImage(named: "headerIsOpen"), for: .normal)
```
Comment (Lomovsky-Alex) [inline Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift]: You need to move button initial setup to separate method called `setupUI` or something like that

File: Test Project#1 Five Systems/ViewControllers/TaskViewController.swift [hunk 3]
```diff
@@ -77,12 +81,22 @@ extension TaskViewController: UITableViewDelegate, UITableViewDataSource {
...
     }
     
-    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
         guard sectionItems.count - 1 >= section else { return nil }
-        return sectionItems[section].title
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: Constants.headerName ) as? TaskDateHeader else { return UIView()}
+        let object = sectionItems[section]
+        header.setUp(object: object)
+//        header.showTasksButton.tag = section
```
Comment (Lomovsky-Alex) [inline Test Project#1 Five Systems/ViewControllers/TaskViewController.swift]: Remove useless code

File: Test Project#1 Five Systems/ViewControllers/TaskViewController.swift [hunk 4]
```diff
@@ -77,12 +81,22 @@ extension TaskViewController: UITableViewDelegate, UITableViewDataSource {
...
+//        header.showTasksButton.addTarget(self, action: #selector(self.openSection), for: .touchUpInside)
+        return header
     }
+//
+//    @objc func openSection(button: UIButton ) {
```
Comment (Lomovsky-Alex) [inline Test Project#1 Five Systems/ViewControllers/TaskViewController.swift:130]: Remove useless code

File: Test Project#1 Five Systems/ViewControllers/SettingsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    
+    let user: UserModel? = CoreDataManager.shared.fetchCurrentUser()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupLabels()
+    }
+    
+    private func setupLabels() {
+        guard let user = user else { return }
```
Comment (Lomovsky-Alex) [inline Test Project#1 Five Systems/ViewControllers/SettingsViewController.swift:16]: No need for guard check because UILabel's `text` property is optional

**Review by Lomovsky-Alex — CHANGES_REQUESTED** [review]

File: Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift (hunk 2 above)
Comment (dkmityuk) [reply Test Project#1 Five Systems/Custom Headers & Sections/TaskDateHeader.swift]: I just removed 25 row because we can not use it on this application

**Review by Lomovsky-Alex — APPROVED** [review]

**Pull request merged**